src/main/java/org/reactome/release/dataexport/FTPFileUploader.java

```diff
@@ -0,0 +1,468 @@
+	 *
+	 * @return FTPClient to the FTP Server
+	 */
+	protected FTPClient getFtpClientToServer() {
+		if (this.ftpClientConnectionToServer == null) {
+			this.ftpClientConnectionToServer = new FTPClient();
+		}
+		return this.ftpClientConnectionToServer;
+	}
+
+	/**
+	 * Returns the path of the local output file directory.
+	 *
+	 * @return Local output file directory
+	 */
+	protected String getLocalOutputDirectoryPath() {
+		final String defaultOutputDirectoryPath = "output";
+		return getProps().getProperty("outputDir", defaultOutputDirectoryPath);
+	}
+
+	/**
+	 * Returns the current Reactome release version (as defined by the configuration file in this project).
+	 *
+	 * @return Reactome release version number
+	 */
+	protected int getReactomeReleaseVersion() {
+		return Integer.parseInt(getProps().getProperty("reactomeVersion"));
+	}
+
+	/**
+	 * Returns the properties object containing values required for connection and interaction between the local
+	 * system and the FTP Server.
+	 *
+	 * @return Properties object containing values for interaction between the local system and with the FTP Server
+	 */
+	protected Properties getProps() {
+		return this.props;
+	}
+
+	/**
+	 * Returns the hostname URL of the FTP Server.
+	 *
+	 * @return Hostname URL of the FTP Server
+	 */
+	protected abstract String getServerHostName();
+
+	/**
+	 * Returns the username used to log in to the FTP Server.
+	 *
+	 * @return Username to log in to the FTP Server
+	 * @see #getPassword()
+	 */
+	abstract String getUserName();
+
+	/**
+	 * Returns the password used to log in to the FTP Server for the username returned by the method getUserName.
+	 *
+	 * @return Password to log in to the FTP Server
+	 * @see #getUserName()
+	 */
+	abstract String getPassword();
+
+	/**
+	 * Returns the path of the Reactome specific directory on the FTP Server.
+	 *
+	 * @return Path of the Reactome specific directory on the FTP Server
+	 */
+	abstract String getReactomeDirectoryPathOnFTPServer();
+
+	private void throwIllegalStateExceptionUnlessRequiredPropsPresent() {
+		List<String> missingRequiredProperties =
+			getRequiredProperties()
+				.stream()
+				.filter(requiredProperty -> !props.containsKey(requiredProperty))
+				.collect(Collectors.toList());
+
+		if (!missingRequiredProperties.isEmpty()) {
+			throw new IllegalStateException(
+				"The following required properties are missing from the properties object: " +
+					missingRequiredProperties
+			);
+		}
+	}
+
+	private boolean uploadFileToServer(String fileToUpload) throws IOException {
+		getLogger().info(
+			"Uploading file '" + fileToUpload + "' to server " + getServerHostName()
+		);
+
+		InputStream fileToUploadInputStream = new FileInputStream(fileToUpload);
+
+		if (getFtpClientToServer().storeFile(fileToUpload, fileToUploadInputStream)) {
+			getLogger().info(
+				"Successfully uploaded '" + fileToUpload + "' to server " +
+					getServerHostName()
+			);
```

Comment:
4 lines for a single logging statement?

Reply:
I've reworked this as well using the `{}` placeholders and will upload a commit soon.  I want to go through the project and make sure the formatting of log statements is consistent.

Reply:
Done: https://github.com/reactome/release-data-exporter/pull/4/commits/6e29c856511fee1cb5c5329b9d8706a9cc10fe69